File: Programming Basics/Programming Basics - C#/Exams/Programming Basics Exam - 17 July 2016/Exam - 17 July 2016/5. Diamond/Diamond.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _5.Diamond
{
    class Diamond
    {
        static void Main(string[] args)
        {
            int n = int.Parse(Console.ReadLine());

            int dots = n;
            int stars = (5 * n) - ((2 * n) + 2);

            for (int first = 1; first <= n; first++)
            {
                if (first == 1)
                {
                    Console.Write(new string('.', dots));
                    Console.Write("*");
                    Console.Write(new string('*', stars));
                    Console.Write("*");
                    Console.WriteLine(new string('.', dots));
                    dots--;
                    stars += 2;
                }

                else
                {
                    Console.Write(new string('.', dots));
                    Console.Write("*");
                    Console.Write(new string('.', stars));
                    Console.Write("*");
                    Console.WriteLine(new string('.', dots));
                    dots--;
                    stars += 2;
                }
            }
            Console.WriteLine(new string('*', 5 * n));

            int firstDots = 1;
            int centralDots = (5 * n) - 4;
            for (int second = 1; second <= 2 * n + 1; second++)
            {
                if (second < 2 * n + 1)
                {
                    Console.Write(new string('.', firstDots));
                    Console.Write("*");
                    Console.Write(new string('.', centralDots));
                    Console.Write("*");
                    Console.WriteLine(new string('.', firstDots));
                    firstDots++;
                    centralDots -= 2;
                }

                else
                {
                    Console.Write(new string('.', firstDots));
                    Console.Write("*");
                    Console.Write(new string('*', centralDots));
                    Console.Write("*");
                    Console.WriteLine(new string('.', firstDots));
                    firstDots++;
                    centralDots -= 2;
                }
            }
        }
    }
}

File: Programming Basics/Programming Basics - JavaScript/Exams/Programming Basics Exam - 17 July 2016/5. Diamond/Diamond.js
function drowDiamond([arg1]) {
    let n = Number(arg1);

    let dots = n;
    let stars = (5 * n) - ((2 * n) + 2);

    for (let first = 1; first <= n ; first++)
    {
        if (first == 1) {
            console.log(".".repeat(dots) + "*" + "*".repeat(stars) + "*" + ".".repeat(dots));
            dots--;
            stars += 2;
        }

        else {
            console.log(".".repeat(dots) + "*" + ".".repeat(stars) + "*" + ".".repeat(dots));
            dots--;
            stars += 2;
        }
    }

    console.log("*".repeat(5 * n));

    let firstDots = 1;
    let centralDots = (5 * n) - 4;

    for (let second = 1; second <= 2 * n + 1; second++)
    {
        if (second < 2 * n + 1) {
            console.log(".".repeat(firstDots) + "*" + ".".repeat(centralDots) + "*" + ".".repeat(firstDots));
            firstDots++;
            centralDots -= 2;
        }

        else {
            console.log(".".repeat(firstDots) + "*" + "*".repeat(centralDots) + "*" + ".".repeat(firstDots));
        }
    }
}

/* Uncomment to test:
drowDiamond(["6"]);
    */